### PR TITLE
Implement geo query and dataset

### PR DESCRIPTION
**Pull request by 64json**
- Implement geo query.
- Implement geo dataset.
- Add tests for both.

### CODE REVIEW

File: src/models/data-cube/filters.ts [hunk 1]
```diff
@@ -26,3 +26,15 @@ export function inOneOfDateRanges(dateRanges: [Date, Date][], rangeOptions: Rang
...
   };
 }
+
+export function inCities(cityIds: string[]): Filter {
+  return (categories: Category[]) => {
+    const cityIndex = categories.findIndex(
+      category => category.name === 'city',
+    );
+    return (row: Row) => {
+      const cityId = row.header[cityIndex] as string;
```
Comment (Awk34) [inline src/models/data-cube/filters.ts:36]: why typecast? Should you coerce any numbers into strings instead?

File: src/datasets/queries/geo.query.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    const rows = dataCube.getDataFor({
+      categoryNames: [cityCategoryName],
+      measureNames,
+      filters: [dateFilter, cityFilter].filter((filter): filter is Filter => filter !== undefined),
+      sortBy: [dateCategoryName],
+    });
+
+    // merge rows to the given unit (e.g., merge city rows to country rows)
+    const rowGroups: Record<string, MeasureValues> = {};
```
[the remark this answers is not in the document]
Comment (64json) [reply src/datasets/queries/geo.query.ts:72]: `Map.get` returns `T | undefined`, which requires me to have non-undefined assertion every time while it is actually never `undefined` in the above case.

File: src/models/data-cube/filters.ts (hunk 1 above)
Comment (64json) [reply src/models/data-cube/filters.ts:36]: `DataCube` is designed to have `string | number` as a header, but city header is always `string`.

File: src/datasets/queries/geo.query.spec.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+          id: easternEuropeSubcontinentId,
+        },
+        unit: TerritoryLevel.CITY,
+      });
+      expect(geoData.length > 0).toBeTrue();
+      expect(geoData.every(geoDatum => cities[geoDatum.id].subcontinentId === easternEuropeSubcontinentId)).toBeTrue();
+    });
+
+    it('Oceania (Continent)', () => {
+      const oceaniaContinentId = '009';
```
[the remark this answers is not in the document]
Comment (64json) [reply src/datasets/queries/geo.query.spec.ts:109]: I'll reflect these changes in #71 to avoid merge conflicts.

**Pull request merged**